### PR TITLE
SOCRadar

### PR DESCRIPTION
**Pull request by SOCRadar**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
SOCRadar contribution pack has been created after local tests are done. Initially the contribution pack includes 2 integrations:
- SOCRadar Incidents (Fetches incidents)
- SOCRadar Threat Fusion (Indicator enrichment by reputation query)

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by marcellarichmond** [comment]
@mchasepan fyi

File: Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+                severity=severity,
+                incident_main_type=incident_main_type,
+                incident_sub_type=incident_sub_type,
+            )
+
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+
+        elif command == 'socradar-mark-incident-fp':
+            incident_id = demisto.args().get('socradar_incident_id')
```
Comment (Shellyber) [inline Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py:514]: Please move the arguments to the command itself (like you did in the ThreatFusion integration

File: Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+            comments = demisto.args().get('comments')
+            return_results(
+                mark_incident_as_fp_command(
+                    client=client,
+                    incident_id=incident_id,
+                    comments=comments
+                )
+            )
+        elif command == 'socradar-mark-incident-resolved':
+            incident_id = demisto.args().get('socradar_incident_id')
```
Comment (Shellyber) [inline Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py:524]: same as above

**Review by Shellyber — COMMENTED** [review]
Thank you for your contribution! 
It looks great! 👍 

Please see my comments.
I also highly recommend that you implement in the ThreatFusion integration the generic reputation commands like `!ip`, `!Domain` `!File`
You can read about it in the following documentation: https://xsoar.pan.dev/docs/integrations/generic-commands-about

File: Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+                    severity: List[str], incident_main_type: Optional[str], incident_sub_type: Optional[str]
+                    ) -> Tuple[Dict[str, int], List[dict]]:
+    """This function retrieves new incidents every interval (default is 1 minute).
+
+    This function has to implement the logic of making sure that incidents are
+    fetched only onces and no incidents are missed. By default it's invoked by
+    XSOAR every minute. It will use last_run to save the timestamp of the last
+    incident it processed. If last_run is not provided, it should use the
+    integration parameter first_fetch_time to determine when to start fetching
+    the first time.
```
[the remark this answers is not in the document]
Comment (SOCRadar) [reply Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py]: done

File: Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py (hunk 1 above)
Comment (SOCRadar) [reply Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py]: done

File: Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py (hunk 2 above)
Comment (SOCRadar) [reply Packs/SOCRadar/Integrations/SOCRadarIncidents/SOCRadarIncidents.py]: done

**Comment by SOCRadar** [comment]
Thank you for your review. We made the necessary changes according to your feedback. 

As you highly recommended, the generic reputation commands are implemented in the ThreatFusion integration. Actually, ThreatFusion integration already had the reputation commands which are functionally equivalent with generic reputation commands. So we decided to change the current commands (socradar-score-ip, socradar-score-domain and socradar-score-hash) with the generic commands (ip, domain and file). We understand the point of generalization of commands across similar integrations so we think with the updated version of ThreatFusion integration, users will be more comfortable by using the (possibly) already known reputation commands.

We can schedule a meeting for the demo any day you want. Just want to let you know that our timezone is GMT+2.

**Comment by Shellyber** [comment]
@SOCRadar I saw the change for ThreatFusion integration. I think that you should leave the names for the commands as you did before, and then implement also the Generic commands `!Ip, !Domain !File` with our [context standard](https://xsoar.pan.dev/docs/integrations/context-standards-mandatory). This will work the best as intended.

**Comment by SOCRadar** [comment]
@Shellyber Alright, I will revert the deleted commands in the last commits (socradar-score-ip, socradar-score-domain and socradar-score-hash) while keeping the newly added generic commands (ip, domain and file). So eventually the ThreatFusion integration will include 6 commands; 

- socradar-score-ip
- socradar-score-domain
- socradar-score-hash
- ip
- domain
- file

About including the DbotScore in generic commands, I believe that it is already included in the output context in a way that is being described in the [DbotScore documentation](https://xsoar.pan.dev/docs/integrations/dbot).

_```The DBot score must be at the root level of the context...```_

I used the **build_dbot_entry** function from **CommonServerPython** in order to create the proper DbotScore entry and then it is being passed into the context as a root level key: value pair. Please see code block from **ip_command** function:

https://github.com/demisto/content/blob/47e721b3491b074101970ce3bfd2f3b205154641/Packs/SOCRadar/Integrations/SOCRadarThreatFusion/SOCRadarThreatFusion.py#L325-L329

Is there something that I am missing here? Should I include the DbotScore in a different way?

**Comment by Shellyber** [comment]
@SOCRadar Exactly! The commands looks great.
And yes, you are using this correctly, just don't forget to add the dbotscore context outputs to the yml as well (It is missing at the moment)
```
        - contextPath: DBotScore.Indicator
          description: The indicator that was tested.
          type: String
        - contextPath: DBotScore.Score
          description: The actual score.
          type: Number
        - contextPath: DBotScore.Type
          description: The indicator type.
          type: String
        - contextPath: DBotScore.Vendor
          description: The vendor used to calculate the indicator score.
```

**Comment by SOCRadar** [comment]
@Shellyber Ahh yes! dbotscore context outputs were missing from .yml file. I added the previously deleted command functions. Now ThreatFusion integration has 6 commands like I mentioned above. Documentation and .yml file have been modified accordingly.

**Comment by Shellyber** [comment]
@SOCRadar Last thing that is missing is the [context standard](https://xsoar.pan.dev/docs/integrations/context-standards-mandatory#file) outputs from the yml file.
Please add those and let's schedule a demo, How about tomorrow at noon? around 14:00?

**Comment by SOCRadar** [comment]
@Shellyber Done! Context standard outputs (those are being produced by the commands) have been added both to documentation and yml file of ThreatFusion integration.

Is it possible to schedule the demo any time on wednesday instead of tomorrow? Lets say around 10:00 GMT+2?

**Comment by Shellyber** [comment]
@SOCRadar Great. I've sent an invite in the mail. Let me know if the time works for you.

**Comment by SOCRadar** [comment]
@Shellyber That's great! I will prepare some use cases beforehand to show the current functionality of the integrations just like in the demo video. Hope to see you on wednesday.

**Comment by evisochek** [comment]
As we discussed in the demo please add the following:
1) Add task that checks if the integration is available
2) The entity enrichment playbook execution should be defeminated by playbook input  

I sent you an email with all relevant instruction.

**Comment by SOCRadar** [comment]
@evisochek Required modifications have been made and all necessary files have been updated including playbook image and readme file.

@Shellyber Discussed modifications have been applied as follows:

- SOCRadar Incidents integration's commands' error handling has been extended. socradar_incident_id parameter will be accepted only as integer (if not throws error). 

image

- SOCRadar Incidents integration's socradar-mark-incident-fp and socradar-mark-incident-resolved commands' socradar_incident_id parameter has to match with the company of the used API key for the integration in SOCRadar platform otherwise throws an error.

image

- Context standards of SOCRadar ThreatFusion integration's domain enrichment command have been extended with additional fields.

**Review by evisochek — APPROVED** [review]

**Review by Shellyber — APPROVED** [review]

**Pull request merged**